homeassistant/helpers/translation.py

```diff
@@ -323,3 +337,111 @@ async def async_get_translations(
     for entry in cached:
         result.update(entry)
     return result
+
+
+async def async_load_translations(
+    hass: HomeAssistant,
+    language: str,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> None:
+    """Prime backend translation cache.
+
+    If integrations are not specified, translation cache is primed for all loaded integrations.
+    """
+    lock = hass.data.setdefault(TRANSLATION_LOAD_LOCK, asyncio.Lock())
+
+    components = _async_get_components_to_load(hass, category, integrations)
+
+    async with lock:
+        if TRANSLATION_FLATTEN_CACHE in hass.data:
+            cache = hass.data[TRANSLATION_FLATTEN_CACHE]
+        else:
+            cache = hass.data[TRANSLATION_FLATTEN_CACHE] = _TranslationCache(hass)
+        await cache.async_load(language, components)
+
+
+@callback
+def async_get_cached_translations(
+    hass: HomeAssistant,
+    language: str,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> dict[str, Any]:
+    """Return all cached backend translations.
+
+    If integrations specified, return translations for them.
+    Otherwise default to loaded integrations.
+    """
+    components = _async_get_components_to_load(hass, category, integrations)
```

Comment:
`_async_get_components_to_load` is a misleading name since we don't load here. Maybe just `_async_get_components` is enough?

Reply:
Renamed